### ISSUE TITLE
Writing to home dir is crashing my submission.

### ISSUE THREAD

**Issue by zakmachachi** [Matgenix/jobflow-remote#81]
Dear developers, 

My cluster does not allow me to write to the home dir from the compute node. I am having some weird issues:

```
Lmod is automatically replacing "cce/15.0.0" with "gcc/11.2.0".


Due to MODULEPATH changes, the following have been reloaded:
  1) cray-mpich/8.1.23


The following have been reloaded with a version change:
  1) gcc/11.2.0 => gcc/10.3.0

Matplotlib created a temporary cache directory at /tmp/matplotlib-e50zqfwa because the default path (/home/e89/e89/zem/.config/matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.
An Error occurred during the command execution: PermissionError [Errno 13] 
Permission denied: '/home/e89'
```

When I set the `MPLCONFIGDIR` variable in the bash script, the warning from MPL disappears but I still get this Permission Denied error. I have set the work dir to the correct place so i am unsure why it is writing to the home dir. I have built the packages using a Python env and also using Conda so I am really stuck here. 

I can run the script from the command line using `jf execution run /work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1` and it runs fine it seems. 

My generated submission is as follows:

```
#!/bin/bash

#SBATCH --partition=standard
#SBATCH --job-name=add
#SBATCH --nodes=1
#SBATCH --time=00:05:00
#SBATCH --account=e89-oxc
#SBATCH --mail-user=redacted
#SBATCH --mail-type=ALL
#SBATCH --output=/work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1/queue.out
#SBATCH --error=/work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1/queue.err
#SBATCH --qos=short
cd /work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1
module load cray-python
module switch PrgEnv-cray PrgEnv-gnu
source /work/e89/e89/zem/jf_remote_env/bin/activate
which python
module load vasp/6

jf execution run /work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1
```

**Comment by ml-evs** [comment]
The matplotlib warning might be a red herring (it seems to suggest its happy to just write to `/tmp` and carry on). Can you post anymore about the workflow you are using and your jobflow-remote config? Assuming your jobflow-remote work_dir is set to `/work/e89/...` etc then there is no reason why jobflow-remote itself would try writing to home, but a workflow might... Could you also try running `jf project check` and make sure your current config passes all the checks?

**Comment by zakmachachi** [comment]
The workflow is a simple test:

```
from jobflow_remote.utils.examples import add
from jobflow_remote import submit_flow
from jobflow import Flow

job1 = add(1, 2)
job2 = add(job1.output, 2)

flow = Flow([job1, job2])

resources = {"nodes": 1, "partition": "standard", "qos": "short" , "time": "00:05:00", "mail_user": "redacted", "mail_type": "ALL", "account": "e89-oxc"}

print(submit_flow(flow, worker="example_worker", resources=resources, project="test_project")) 
```

My config is as follows:

```
name: test_project
base_dir: .jfremote/test_project
tmp_dir: .jfremote/test_project/tmp
log_dir: .jfremote/test_project/log
daemon_dir: .jfremote/test_project/daemon
log_level: debug
runner:
  delay_checkout: 30
  delay_check_run_status: 30
  delay_advance_status: 30
  delay_refresh_limited: 600
  delay_update_batch: 60
  lock_timeout: 86400
  delete_tmp_folder: true
  max_step_attempts: 3
  delta_retry:
  - 30
  - 300
  - 1200
workers:
  example_worker:
    type: remote
    scheduler_type: slurm
    work_dir: /work/e89/e89/zem/jf_out
    resources:
    pre_run: |
       module load cray-python
       module switch PrgEnv-cray PrgEnv-gnu
       source /work/e89/e89/zem/jf_remote_env/bin/activate
       which python
       module load vasp/6
    post_run:
    timeout_execute: 120
    max_jobs: 10
    batch:
    host: login.archer2.ac.uk
    user: zem
    port: 
    password:
    key_filename: /u/vld/sedm6197/.ssh/id_rsa
    passphrase:
    gateway:
    forward_agent:
    connect_timeout:
    connect_kwargs:
    inline_ssh_env:
    keepalive: 60
    shell_cmd: bash
    login_shell: true
    interactive_login: false
queue:
  store:
    type: MongoStore
    host: redacted
    database: atomate2
    username: redacted
    password: redacted
    collection_name: jobs_remote
  flows_collection: flows_remote
  auxiliary_collection: jf_auxiliary
  db_id_prefix:
exec_config: {}
jobstore:
  docs_store:
    type: MongoStore
    database: atomate2
    host: redacted
    port: 27017
    username: redacted
    password: redacted
    collection_name: outputs_remote
  additional_stores:
    data:
      type: GridFSStore
      database: atomate2
      host: redacted
      port: 27017
      username: redacted
      password: redacted
      collection_name: outputs_blobs_remote
metadata:
```

jf project check passes with the following:

```
(BAM-remote) [sedm6197@chromite jf-remote]$ jf project check
The selected project is test_project from config file /u/vld/sedm6197/.jfremote/test_project.yaml
✓ Worker example_worker
✓ Jobstore
✓ Queue store
```

**Comment by gpetretto** [comment]
Thanks for reporting this. 
It could be some hidden attempt to create a standard folder. One example could be this: https://github.com/Matgenix/jobflow-remote/blob/5da2c9e5f3570662f0e902e36f0a985c14464ec5/src/jobflow_remote/config/manager.py#L70

Can you try creating the ~/.jfremote folder and trying again? (actually it could be that the folder has been created if you have run `jf execution run /work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1` directly on the frontend. In case you didn't try after that, can you try again?)

If it happens again, can you try resubmitting the script that you posted in the first message modifying it like this?
Replace the line 
`jf execution run /work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1`
with
`jf -fe execution run /work/e89/e89/zem/jf_out/02/a0/e1/02a0e183-d1cc-4b4a-a818-94c6edf96f42_1`

This would hopefully tell us where the error is coming from exactly.

I will make this the default and also try to get rid of the makedirs_p in the ConfigManager.

**Comment by zakmachachi** [comment]
Sorry for the delayed response.. So I always had a .jfremote folder in my work on the HPC and on my local machine. Do I need to also make one in my home dir on the HPC also? In any case, the cluster has been having connectivity issues the past few days so I am unable to test until it's fixed. I'll report back once it's up and running again. Many thanks!

**Comment by gpetretto** [comment]
The `.jfremote` does not need ti be present on the cluster. It is just that the code tries to create it anyway. I removed this mkdir, but the changes still need to merged.

**Comment by gpetretto** [comment]
This should have been solved by #87. Feel free to reopen the issue if the problem shows up again.
